File: C++/PetShop.cpp
#include <iostream>
#include <string>

using namespace std;

// Mendefinisikan class PetShop
class PetShop {
    // Bagian private: menyimpan data yang hanya bisa diakses di dalam class
    private:
    int id;            // Menyimpan ID produk
    string name;       // Menyimpan nama produk
    string category;   // Menyimpan kategori produk
    int price;         // Menyimpan harga produk
    
    // Bagian public: menyimpan method yang bisa diakses dari luar class
    public:
    // Constructor default: membuat objek dengan nilai awal
    PetShop() {
        this->id = -1;         // Set ID default ke -1
        this->name = "";       // Set nama default ke string kosong
        this->category = "";   // Set kategori default ke string kosong
        this->price = 0;       // Set harga default ke 0
    }

    // Constructor dengan parameter: membuat objek dengan nilai yang ditentukan
    PetShop(int id, string name, string category, int price) {
        this->id = id;               // Set ID sesuai parameter
        this->name = name;           // Set nama sesuai parameter
        this->category = category;   // Set kategori sesuai parameter
        this->price = price;         // Set harga sesuai parameter
    }

    // Method untuk mengubah data produk
    void setData(int newId, string newName, string newCategory, int newPrice) {
        this->id = newId;               // Update ID
        this->name = newName;           // Update nama
        this->category = newCategory;   // Update kategori
        this->price = newPrice;         // Update harga
    }
    
    // Getter untuk mendapatkan ID produk
    int getId() {
        return this->id;
    }
    
    // Getter untuk mendapatkan nama produk
    string getName() {
        return this->name;
    }
    
    // Getter untuk mendapatkan kategori produk
    string getCategory() {
        return this->category;
    }
    
    // Getter untuk mendapatkan harga produk
    int getPrice() {
        return this->price;
    }

    // Destructor: dipanggil saat objek dihapus
    ~PetShop() {
    }
};

File: PHP/PetShop.php
<?php

class PetShop {
    // Properti private untuk menyimpan informasi produk
    private $id;         // Menyimpan ID produk
    private $name;       // Menyimpan nama produk
    private $category;   // Menyimpan kategori produk
    private $price;      // Menyimpan harga produk
    private $image;      // Menyimpan path gambar produk
    
    // Constructor untuk inisialisasi nilai default
    public function __construct() {
        $this->id = -1;          // Set ID default ke -1
        $this->name = "";        // Set nama default ke string kosong
        $this->category = "";    // Set kategori default ke string kosong
        $this->price = 0;        // Set harga default ke 0
        $this->image = "";       // Set image path default ke string kosong
    }
    
    // Method untuk mengatur semua data produk sekaligus
    public function setData($id, $name, $category, $price, $image = "") {
        $this->id = $id;                 // Set ID produk
        $this->name = $name;             // Set nama produk
        $this->category = $category;     // Set kategori produk
        $this->price = $price;           // Set harga produk
        $this->image = $image;           // Set path gambar produk
    }
    
    // Getter methods untuk mengakses data private
    public function getId() {
        return $this->id;        // Mengembalikan ID produk
    }
    
    public function getName() {
        return $this->name;      // Mengembalikan nama produk
    }
    
    public function getCategory() {
        return $this->category;  // Mengembalikan kategori produk
    }
    
    public function getPrice() {
        return $this->price;     // Mengembalikan harga produk
    }

    public function getImage() { 
        return $this->image;     // Mengembalikan path gambar produk
    }
}

// Cek dan mulai session jika belum dimulai
if (session_status() == PHP_SESSION_NONE) {
    session_start();
}

// Inisialisasi array data di session jika belum ada
if (!isset($_SESSION['petshop_data'])) {
    $_SESSION['petshop_data'] = array();     // Array untuk menyimpan objek PetShop
    $_SESSION['item_count'] = 0;             // Counter jumlah item
}

// Membuat direktori uploads jika belum ada
$uploadDir = 'uploads/';
if (!file_exists($uploadDir)) {
    mkdir($uploadDir, 0777, true);           // Membuat folder dengan permission penuh
}

?>